### PR TITLE
[prometheus] moving node-exporter out of prometheus

### PR DESCRIPTION
**Pull request by zanac1986**
#### What this PR does / why we need it

Move node-exporter the embedded chart back to its parent. As mentioned here:

https://github.com/prometheus-community/helm-charts/pull/2627#issuecomment-1300167738
https://github.com/prometheus-community/helm-charts/pull/2627#issuecomment-1300192978
https://github.com/prometheus-community/helm-charts/issues/2654#issuecomment-1306468145

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus/templates/NOTES.txt [hunk 1]
```diff
@@ -70,13 +70,13 @@ Get the Alertmanager URL by running these commands in the same shell:
 {{- end }}
 {{- end }}
 
-{{- if .Values.nodeExporter.podSecurityPolicy.enabled }}
+{{- if and (index .Values "prometheus-node-exporter" "enabled") (index .Values "prometheus-node-exporter" "rbac" "pspEnabled") }}
```
Comment (monotek) [inline charts/prometheus/templates/NOTES.txt:73]: imho podsecuritypolicy should be removed completly, as it's already deprecated.

File: charts/prometheus/values.yaml [hunk 2]
```diff
@@ -481,39 +477,148 @@ kubeStateMetrics:
...
+    # https://kubernetes.io/docs/concepts/storage/volumes/#mount-propagation
+    mountPropagation: HostToContainer
 
-  ## node-exporter container image
-  ##
   image:
     repository: quay.io/prometheus/node-exporter
-    tag: v1.3.1
+    # Overrides the image tag whose default is {{ printf "v%s" .Chart.AppVersion }}
+    tag: "v1.3.1"
```
Comment (monotek) [inline charts/prometheus/values.yaml:504]: not sure if we need to define the tag by default.
will already be updated through chart version.

File: charts/prometheus/values.yaml [hunk 3]
```diff
@@ -523,120 +628,157 @@ nodeExporter:
...
+    # Fix for very slow GKE cluster upgrades
+    cluster-autoscaler.kubernetes.io/safe-to-evict: "true"
+
+  # Extra labels to be added to node exporter pods
+  podLabels: {}
 
-  ## Additional node-exporter container arguments
-  ##
-  extraArgs: {}
+  ## set to true to add the release label so scraping of the servicemonitor with kube-prometheus-stack works out of the box
```
Comment (monotek) [inline charts/prometheus/values.yaml:655]: imho not needed as kube kube-prometheus-stack specific

File: charts/prometheus/Chart.yaml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v2
 name: prometheus
 appVersion: 2.39.1
-version: 15.18.0
+version: 15.19.0
```
Comment (monotek) [inline charts/prometheus/Chart.yaml:4]: I think thats worth a major version update

**Review by monotek — CHANGES_REQUESTED** [review]
Has the uprade path been tested?
Please add some info about it in the Readme.

Please remove all unchanged config from values.yaml.

File: charts/prometheus/values.yaml (hunk 2 above)
Comment (zanac1986) [reply charts/prometheus/values.yaml:504]: This one is to maintain the compatibility with previous embedded chart.

**Comment by zanac1986** [comment]
> Has the uprade path been tested? Please add some info about it in the Readme.

The code status only reaches around 95%+ similarity with `helm template`. I don't have very deep test cases, and would like if somebody could take over that part of jobs.

File: charts/prometheus/values.yaml (hunk 2 above)
Comment (monotek) [reply charts/prometheus/values.yaml:504]: i would still remove it as a chart update of proemtheus would keep the old version of node-exporter otherwise.

**Review by monotek — CHANGES_REQUESTED** [review]
please add update instructions and whats changed to upgrading section of readme

File: charts/prometheus/README.md [hunk 5]
```diff
@@ -56,6 +57,10 @@ helm upgrade [RELEASE_NAME] [CHART] --install
 
 _See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documentation._
 
+### To 16.0
+
+Start from 16.0, prometheus moves all embedded charts out of prometheus chart umbrella back to its parent. Version 16.0.0 moves node-exporter to external promtheus-node-exporter. If you've made some manual changes, please diff the old (`nodeExporter`) and the new (`prometheus-node-exporter`) configuration section in values.yaml. Uninstalling the old chart release and then reinstalling the new chart release is recommended.
```
Comment (monotek) [inline charts/prometheus/README.md:62]: I guess "uninstalling" is not the best idea, as the prometheus chart default is to use a deployment and the pvc would likely be deleted too, so you would loose all data.

So i would try too give non destructive upgrade instructions instead. Best thin would be to try it out once and describe any special steps which might have to be done.

**Review by monotek — CHANGES_REQUESTED** [review]

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus/README.md [hunk 6]
```diff
@@ -59,7 +59,16 @@ _See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documen
 
 ### To 16.0
 
-Start from 16.0, prometheus moves all embedded charts out of prometheus chart umbrella back to its parent. Version 16.0.0 moves node-exporter to external promtheus-node-exporter. If you've made some manual changes, please diff the old (`nodeExporter`) and the new (`prometheus-node-exporter`) configuration section in values.yaml. Uninstalling the old chart release and then reinstalling the new chart release is recommended.
+Start from 16.0, prometheus moves all embedded charts out of prometheus chart umbrella back to its parent. Version 16.0.0 moves node-exporter to external promtheus-node-exporter. If you've made some manual changes, please diff the old (`nodeExporter`) and the new (`prometheus-node-exporter`) configuration section in values.yaml.
```
Comment (monotek) [inline charts/prometheus/README.md:62]: Starting from version 16.0 embedded services (like alertmanager, node-exporter etc.) are moved out of Prometheus chart and the respecting charts from this repo are used as dependencies. Version 16.0.0 moves node-exporter service to [prometheus-node-exporter chart](https://github.com/prometheus-community/helm-charts/tree/main/charts/prometheus-node-exporter). If you've made some config changes, please check the old `nodeExporter` and the new `prometheus-node-exporter` configuration section in values.yaml for differences.

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus/README.md [hunk 7]
```diff
@@ -96,7 +105,7 @@ Assuming you have an existing release of the prometheus chart, named `prometheus
     alertmanager.yml: ""
   kubeStateMetrics:
     enabled: false
-  nodeExporter:
+  nodeExporter:                       # <= Change to prometheus-node-exporter after version 16.x.x
```
[the remark this answers is not in the document]
Comment (zanac1986) [reply charts/prometheus/README.md]: I drop the change for this one because it seems that for prometheus itself upgrade, not chart related.

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**